File: src/kbs/baekjoon/silver/Baekjoon6588.java
package kbs.baekjoon.silver;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Baekjoon6588 {
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		boolean primeNum[]=new boolean[1000001];
		primeNum[0]=true; primeNum[1]=true;
		for(int i=2;i<Math.sqrt(1000000);i++) {
			for(int j=2;i*j<1000000;j++) {
				primeNum[i*j]=true;
			}
		}
		while(true) {
			int n = Integer.parseInt(br.readLine());
			if(n==0) break;
			int a=1;
			for(a=n-1;a>1;a--) {
				if(!primeNum[a] && !primeNum[n-a]) break;
			}
			if(!primeNum[a] && !primeNum[n-a]) {
				System.out.println(n+" = "+(n-a)+" + "+a);
			}else {
				System.out.println("Goldbach's conjecture is wrong.");
			}
		}
	}
}

File: src/ljh/baekjoon/silver/baekjoon_6588.py
# baekjoon 알고리즘 6588
# 2023.09.17 이주현

import sys
import math

### 반례 찾기 -> 아무리봐도 모르겟다..!

check_array = [False, False] + [True]*1000000

for i in range(2, 1000000):
    if check_array[i]:
        for j in range(2*i, 1000000, i):
            check_array[j] = False

while True:
    num = int(sys.stdin.readline())

    if num == 0:
        break

    is_correct = False

    a = num - 2
    b = num - a
    
    while a >= b:

        if check_array[a] and check_array[b]:
            is_correct = True
            print(num , " = " , b , " + " , a)
            break
        a -= 1
        b = num - a

    if not is_correct:
        print("Goldbach's conjecture is wrong.")
